src/librustc/mir/visit.rs

```diff
@@ -737,11 +737,18 @@ macro_rules! make_mir_visitor {
                         self.visit_local(local, context, location);
                     }
                     Place::Base(PlaceBase::Static(static_)) => {
+                        match static_.promoted {
```

Comment:
This match is unnecessary. The type is visited inside `visit_static`

Reply:
I have clubbed both the cases to use `visit_static`. Is this what we want?

Reply:
I think so. Can you check all implementations of `visit_static` to ensure that they don't do things that are wrong for promoteds?